Source/Core/VideoCommon/CPMemory.h

```diff
@@ -255,6 +257,46 @@ struct CPState final
   bool bases_dirty;
   VertexLoaderBase* vertex_loaders[8];
   int last_id;
+
+  CPState & operator=(std::nullptr_t)
+  {
+      memset(array_bases, 0, sizeof(array_bases));
+      memset(array_strides, 0, sizeof(array_strides));
+      matrix_index_a = {0};
+      matrix_index_b = {0};
+      vtx_desc = {0};
+      for(auto &v : vtx_attr)
+      {
+          v = {0};
+      }
+      attr_dirty = {0};
+      bases_dirty = false;
+      last_id = 0;
+      for(auto &l : vertex_loaders)
+      {
+          l = nullptr;
+      }
+
+      return *this;
+  }
+
+  CPState & operator=(const CPState& new_state)
```

Comment:
This is performing the same behavior as the default copy constructor, so I don't see why this whole function needs to be hardcoded in.